cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -0,0 +1,224 @@
+    if (pathPrefix == null) {
+      return localRepoPath;
+    }
+    return localRepoPath.resolve(pathPrefix);
+  }
+
+  /**
+   * Validates the provided configuration.
+   *
+   * @param secureStore         A secure store for fetching credentials if required.
+   * @param sourceControlConfig Configuration for source control operations.
+   * @throws IOException                         when there are File I/O problems.
+   * @throws RepositoryConfigValidationException when provided repository configuration is invalid.
+   */
+  public static void validateConfig(SecureStore secureStore, SourceControlConfig sourceControlConfig) throws
+    IOException, RepositoryConfigValidationException {
+    try {
+      doValidate(secureStore, sourceControlConfig);
+    } finally {
+      deletePathIfExists(sourceControlConfig.getReposBasePath());
+    }
+  }
+
+  /**
+   * Initializes the Git repository by cloning remote.
+   *
+   * @throws GitAPIException               when a Git operation fails.
+   * @throws AuthenticationConfigException when correct authentication credentials are not available.
+   * @throws IOException                   when file or network I/O fails.
+   */
+  @Override
+  protected void startUp() throws Exception {
+    if (git != null) {
+      return;
+    }
+    // Clean up the directory if it already exists.
+    deletePathIfExists(sourceControlConfig.getLocalRepoPath());
+    RepositoryConfig repositoryConfig = sourceControlConfig.getRepositoryConfig();
+    CloneCommand command = createCommand(Git::cloneRepository).setURI(repositoryConfig.getLink())
+                                                              .setDirectory(sourceControlConfig.getLocalRepoPath()
+                                                                                               .toFile());
+
+    String branch = getBranchRefName(repositoryConfig.getDefaultBranch());
+    if (branch != null) {
+      command.setBranchesToClone(Collections.singleton((branch))).setBranch(branch);
+    }
+    git = command.call();
+  }
+
+  @Override
+  protected void shutDown() throws Exception {
+    if (git != null) {
+      git.close();
+    }
+    git = null;
+    deletePathIfExists(sourceControlConfig.getReposBasePath());
+  }
+
+  private static void doValidate(SecureStore secureStore, SourceControlConfig sourceControlConfig) throws IOException {
+    // Validate existence of secret in Secure Store.
+    RepositoryConfig config = sourceControlConfig.getRepositoryConfig();
+    LOG.info("Checking existence of secret in secure store: {}", config.getAuth().getTokenName());
+    try {
+      secureStore.get(sourceControlConfig.getNamespaceID(), config.getAuth().getTokenName()).get();
+    } catch (Exception e) {
+      throw new RepositoryConfigValidationException("Failed to get auth credentials from secure store.", e);
+    }
+
+    // Try fetching heads in the remote repository.
+    FileRepository localRepo;
```

Comment:
You can use the in memory repo instead so that we don't need to deal with the file path creation and deletion.

```
Git git = Git.wrap(new InMemoryRepository.Builder().build());
```

Reply:
Done. Thanks for the idea.